spec/requests/user_request_spec.rb

```diff
@@ -1,90 +1,122 @@
+      end
+      it "should return auth_token" do
+        expect(@post_response_body["data"]["attributes"]).to have_key("authentication_token")
+      end
       
-      # do not include password or api_secret
-      expect(response_body["data"]["attributes"]).not_to (have_key("password") && have_key("api_secret"))
+      it "should not include password or api_secret" do
+        expect(@post_response_body["data"]["attributes"]).not_to (have_key("password") && have_key("api_secret"))
+      end
+    end
 
-      # cannot update as it does not have the password (or a password)
-      patch "/api/v1/users", headers: create_headers("newuser@example.com", auth_token), params: {"api_secret": "new_api_secret"}
-      expect(response).to have_http_status(:bad_request)
+    describe "GET: current user" do
+      before(:each) do
+        get "/api/v1/users", headers: @headers
+        @response_body = get_response_body(response)
+      end
+      it "should still be the same data" do
+        expect(@response_body["data"]["attributes"]).to include("email" => user_params["email"], "api_key" => user_params["api_key"])
+      end
+      it "should return auth token" do
+        expect(@response_body["data"]["attributes"]).to have_key("authentication_token")
+      end
+      it "should not include password or api_secret" do
+        expect(@response_body["data"]["attributes"]).not_to (have_key("password") && have_key("api_secret"))
+      end
     end
 
-    it "DELETE: current user" do
-      _response = create_user(user_params)
-      response_body = get_response_body(_response)
-      auth_token = get_by_attribute(response_body, "authentication_token")
+    describe "PATCH: current user" do
+      new_email = "newuser@example.com"
+      new_api_key = "new_api_key"
+
+      describe "with password" do
+        before(:each) do
+          patch "/api/v1/users", headers: @headers, params: {"email" => new_email, "api_key": new_api_key, "password" => user_params["password"]}
+          @response_body = get_response_body(response)
+        end
+        it "should have new data" do
+          expect(@response_body["data"]["attributes"]).to include("email" => new_email, "api_key" => new_api_key)
+        end
+        it "should return auth token" do
+          expect(@response_body["data"]["attributes"]).to have_key("authentication_token")
+        end
+        it "should not include password or api_secret" do
+          expect(@response_body["data"]["attributes"]).not_to (have_key("password") && have_key("api_secret"))
+        end
+      end
+      
+      describe "without password" do
+        before(:each) do
+          patch "/api/v1/users", headers: @headers, params: {"email" => new_email, "api_key": new_api_key}
+          @response = response
+        end
+        it "should get bad request as password was not provided" do
+          expect(@response).to have_http_status(:bad_request)
+        end
```

Comment:
Aquí la gracia es que el context permita entender porqué falla, en el test sería suficiente que pusieras solamente `it { expect(@response).to have_http_status(:bad_request) }`